controllers/kubeadmconfig_controller.go

```diff
@@ -52,12 +55,13 @@ type KubeadmConfigReconciler struct {
 // SecretsClientFactory define behaviour for creating a secrets client
 type SecretsClientFactory interface {
 	// NewSecretsClient returns a new client supporting SecretInterface
-	NewSecretsClient(client.Client, *capiv1alpha2.Cluster) (corev1.SecretInterface, error)
+	NewSecretsClient(client.Client, *capiv1alpha2.Cluster) (typedcorev1.SecretInterface, error)
 }
 
 // +kubebuilder:rbac:groups=bootstrap.cluster.x-k8s.io,resources=kubeadmconfigs,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=bootstrap.cluster.x-k8s.io,resources=kubeadmconfigs/status,verbs=get;update;patch
 // +kubebuilder:rbac:groups=cluster.x-k8s.io,resources=clusters;machines,verbs=get;list;watch
+// +kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch;create;update;patch
```

Comment:
I think this change is only using get and create in which case this should be scoped down to the minimal necessary subset

Reply:
There is a Create as well, so it's likely it will need all of them (except maybe patch)

Reply:
it is surprising to me that create use list, watch update and maybe patch, but that wouldn't be the first time a k8s client has surprised me 😂

Reply:
The controller runtime will use a cache, so list, watch, and possibly get are highly likely to be used.

As for patch, I don't necessarily *think* it will be used, but I'm not certain.

Reply:
i'm going to say this is resolved, it's fine leaving it as is 🎃